BLL/Services/DrinkSearchService.cs

```diff
@@ -98,39 +97,25 @@ public IEnumerable<Drink> SearchByIngredients(SortedSet<string> ingredientsToSea
             return drinksFound;
         }
 
-        public IQueryable<Drink> SearchByAlcoholContent(bool alcoholics, bool nonAlcoholics, bool optionalAlcoholics, List<Drink> drinks)
+        public IEnumerable<Drink> SearchByAlcoholContent(bool alcoholics, bool nonAlcoholics, bool optionalAlcoholics)
         {
+            var drinks = _repository.GetAllDrinksAsQueryable();
             var contemporaryList = new List<Drink>();
             if (alcoholics)
             {
-                contemporaryList.AddRange(drinks.Where(x => x.AlcoholicInfo == DrinkIsAlcoholic).ToList());
+                contemporaryList.AddRange(drinks.Where(x => x.AlcoholicInfo == DrinkIsAlcoholic));
             }
 
             if (nonAlcoholics)
             {
-                contemporaryList.AddRange(drinks.Where(x => x.AlcoholicInfo == DrinkIsNonAlcoholic).ToList());
+                contemporaryList.AddRange(drinks.Where(x => x.AlcoholicInfo == DrinkIsNonAlcoholic));
             }
 
             if (optionalAlcoholics)
             {
-                contemporaryList.AddRange(drinks.Where(x => x.AlcoholicInfo == DrinkIsOptionalAlcohol).ToList());
+                contemporaryList.AddRange(drinks.Where(x => x.AlcoholicInfo == DrinkIsOptionalAlcohol));
             }
```

Comment:
I've optimized SearchByAlcoholContent according to Patryk's and Wojtek's advices